### PR TITLE
Improve MergeUnitsSorting to have several merge at the same time.

### PR DESCRIPTION
**Pull request by samuelgarcia**
@ferchaure @alejoe91

This is a quick hack to improve MergeUnitsSorting to have several merge at the same time.

This class should be refactor with more option of merging stragetgy. But this is outside the PR.

This is needed for the benchmark correction.

### CODE REVIEW

**Review by ferchaure — COMMENTED** [review]
A few days ago, I was thinking that something like this was necessary 

To be honest I don't really like the current implementation of get_non_duplicated_events... An option could be using https://pypi.org/project/sortednp/
I don't like to add more and more dependencies but be work a lot with sorted arrays

File: spikeinterface/curation/mergeunitssorting.py [hunk 1]
```diff
@@ -28,96 +28,127 @@ class MergeUnitsSorting(BaseSorting):
...
+
+        if new_unit_ids is None:
+            dtype = parents_unit_ids.dtype
             #select new_units_ids grater that the max id, event greater than the numerical str ids 
-            if np.issubdtype(parents_unit_ids.dtype, np.character):
-                new_unit_id = max([0]+[int(p) for p in parents_unit_ids if p.isdigit()])+1
+            if np.issubdtype(dtype, np.character):
+                new_unit_ids = [ f'merge{i}' for i in range(num_merge)]
+                if np.any(np.in1d(new_unit_ids, keep_unit_ids)):
+                    raise VlueError('Unable to find new_unit_ids because it is a string and parents already contain merges')
```
[the remark this answers is not in the document]
Comment (samuelgarcia) [reply spikeinterface/curation/mergeunitssorting.py:54]: We can put back the str to digit old way if you want.
I was just too lazy to do it!!

[the remark this answers is not in the document]
Comment (samuelgarcia) [reply spikeinterface/curation/mergeunitssorting.py]: OK I did it.
@ferchaure can you check it ?

**Comment by samuelgarcia** [comment]
@alejoe91 ready to merge on my side.

File: spikeinterface/curation/mergeunitssorting.py (hunk 1 above)
Comment (ferchaure) [inline spikeinterface/curation/mergeunitssorting.py]: check the VlueError typo

**Review by ferchaure — APPROVED** [review]

File: spikeinterface/curation/mergeunitssorting.py (hunk 1 above)
Comment (alejoe91) [reply spikeinterface/curation/mergeunitssorting.py]: Thanks! @samuelgarcia can you fix?

**Review by alejoe91 — APPROVED** [review]

**Pull request merged**